### PR TITLE
[Merged by Bors] - chore(*): drop some `[Decidable*]` assumptions

### PR DESCRIPTION
**Pull request by urkud**
Found by the linter in #10235

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fintype/Quotient.lean [hunk 1]
```diff
@@ -50,24 +50,26 @@ theorem listChoice_mk {l : List ι} (a : ∀ i ∈ l, α i) : listChoice (S := S
...
-    intro as
-    rw [List.Pi.cons_map a as (fun i ↦ Quotient.mk (S i))]
-    exact f _
+    classical
+      rw [← List.Pi.cons_eta q]
+      induction' List.Pi.head q using Quotient.ind with a
+      refine @list_ind _ (fun q ↦ C (List.Pi.cons _ _ ⟦a⟧ q)) ?_ (List.Pi.tail q)
+      intro as
+      rw [List.Pi.cons_map a as (fun i ↦ Quotient.mk (S i))]
+      exact f _
```
Comment (sgouezel) [inline Mathlib/Data/Fintype/Quotient.lean:67]:
```suggestion
    rw [← List.Pi.cons_eta q]
    induction' List.Pi.head q using Quotient.ind with a
    refine @list_ind _ (fun q ↦ C (List.Pi.cons _ _ ⟦a⟧ q)) ?_ (List.Pi.tail q)
    intro as
    rw [List.Pi.cons_map a as (fun i ↦ Quotient.mk (S i))]
    exact f _
```
I think the current convention is not to indent further after classical

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Data/Fintype/Quotient.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Data/Fintype/Quotient.lean]: @FR-vdash-bot You've added this lemma. Do you care about it being choice free? What do you prefer to make the pending linter happy:

- make it choice dependent as most of `Mathlib`;
- add a choice free version in `Decidable` namespace?

Comment (urkud) [reply Mathlib/Data/Fintype/Quotient.lean]: I'm going to revert this file for now.

**Pull request closed without merging**